code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Food-Drinks.dm

```diff
@@ -244,98 +292,105 @@
 	name = "Capsaicin Oil"
 	id = "capsaicin"
 	description = "This is what makes chilis hot."
+	taste_description = "hot peppers"
+	taste_mult = 1.5
 	reagent_state = LIQUID
 	color = "#B31008"
+	var/agony_dose = 5
+	var/agony_amount = 2
+	var/discomfort_message = "<span class='danger'>Your insides feel uncomfortably hot!</span>"
+	var/slime_temp_adj = 10
 
 /datum/reagent/capsaicin/affect_blood(var/mob/living/carbon/M, var/alien, var/removed)
 	if(alien == IS_DIONA)
 		return
 	M.adjustToxLoss(0.5 * removed)
 
 /datum/reagent/capsaicin/affect_ingest(var/mob/living/carbon/M, var/alien, var/removed)
-	if(alien == IS_DIONA)
+	if(alien == IS_DIONA || alien == IS_MACHINE)
 		return
 	if(ishuman(M))
 		var/mob/living/carbon/human/H = M
-		if(!H.can_feel_pain())
+		if(H.species && (H.species.flags & (NO_PAIN)))
 			return
-
-	if(dose < 5 && (dose == metabolism || prob(5)))
-		M << "<span class='danger'>Your insides feel uncomfortably hot!</span>"
-	if(dose >= 5)
-		M.apply_effect(2, AGONY, 0)
+	if(dose < agony_dose)
+		if(prob(5) || dose == metabolism) //dose == metabolism is a very hacky way of forcing the message the first time this procs
+			M << discomfort_message
+	else
+		M.apply_effect(agony_amount, AGONY, 0)
 		if(prob(5))
-			M.visible_message("<span class='warning'>[M] [pick("dry heaves!","coughs!","splutters!")]</span>", "<span class='danger'>You feel like your insides are burning!</span>")
+			M.custom_emote(2, "[pick("dry heaves!","coughs!","splutters!")]")
+			M << "<span class='danger'>You feel like your insides are burning!</span>"
 	if(istype(M, /mob/living/carbon/slime))
-		M.bodytemperature += rand(10, 25)
+		M.bodytemperature += rand(0, 15) + slime_temp_adj
 	holder.remove_reagent("frostoil", 5)
 
-/datum/reagent/condensedcapsaicin
+/datum/reagent/capsaicin/condensed
```

Comment:
Reagent types are almost never used anywhere, reagents are always referred to by ID. I did check just in case, on Bay at least. I highly doubt Polaris has started using reagent paths for anything.